.github/workflows/daily-snapshots.yml

```
@@ -0,0 +1,182 @@
+name: Daily snapshots
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/daily-snapshots.yml'
+      - 'scripts/create-release.sh'
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  snapshots-5-1:
+    name: Generate 5.1 snapshots
+    runs-on: ubuntu-latest
+    if: (github.event_name == 'schedule' && github.repository == 'phpmyadmin/phpmyadmin') || (github.event_name != 'schedule')
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v3
        with:
            ref: QA_5_1
```
https://github.com/actions/checkout#checkout-a-different-branch